### ISSUE TITLE
feel free to add more examples

### ISSUE THREAD

**Issue by divyanshu-rawat** [divyanshu-rawat/Functional-Programming-JS#1]
feel free to add more examples of functional programming in js.
I will be more than happy to merge your PR.

**Comment by acgrdumlu** [comment]
I just wanna know if you want to me add exapmle that you specified in the README. So do you want me to add some example about closures or something like that.

**Comment by divyanshu-rawat** [comment]
No, I want you to add resources that demonstrate the use of functional programming in javascript.
Like functions in javascript are first class objects and can be passed into other functions.

**Comment by yussan** [comment]
i love to join this repo, hopefully my pull request is suitable

**Comment by Adroit11** [comment]
Would love to join in

**Comment by akshatbhargava123** [comment]
I am a beginner, just PRed for insertion sort. Please give me more work. I would like to do in JavaScript.

**Comment by phatthana** [comment]
I submit a PR about how to compare variable in JS #8

**Comment by Cooper-Cole** [comment]
I would love to add .join because it was something I learned in functional js program and it's pretty simple and easy to use

**Comment by divyanshu-rawat** [comment]
ya @Alyssa-Cooper looking fwd for your PR.

**Comment by Cooper-Cole** [comment]
@divyanshu-rawat I made a pull request with .join!

**Comment by divyanshu-rawat** [comment]
@Alyssa-Cooper  THANKS 👍  let me review it before merging it.

**Comment by sammorrow** [comment]
@divyanshu-rawat made an implementation of spy factory, it's up as a PR now. let me know if it needs any changes.
